src/model.rs

```diff
@@ -74,17 +76,55 @@ impl Collections {
             trip_properties,
             geometries,
             admin_stations,
+            stop_time_headsigns,
         } = c;
         self.contributors.merge(contributors)?;
         self.datasets.merge(datasets)?;
         self.networks.merge(networks)?;
         self.commercial_modes.merge(commercial_modes)?;
         self.lines.merge(lines)?;
         self.routes.merge(routes)?;
-        self.vehicle_journeys.merge(vehicle_journeys)?;
         self.physical_modes.extend(physical_modes);
         self.stop_areas.merge(stop_areas)?;
+
+        fn get_new_idx<T>(
+            old_idx: Idx<T>,
+            old_idx_to_id: &HashMap<Idx<T>, String>,
+            merge_collection: &CollectionWithId<T>,
+        ) -> Option<Idx<T>> {
+            old_idx_to_id
+                .get(&old_idx)
+                .and_then(|id| merge_collection.get_idx(id))
+        }
+
+        let sp_idx_to_id = stop_points.idx_to_id();
+        let vj_idx_to_id = vehicle_journeys.idx_to_id();
+
         self.stop_points.merge(stop_points)?;
+        self.vehicle_journeys.merge(vehicle_journeys)?;
+
+        // Update stop point idx in stop times
+        let mut vjs = self.vehicle_journeys.take();
```

Comment:
That's done on the merged view, it will not work: old vjs will have stop_points on the new stop points.

We need a test that merge 2 models and check that the vj are as expected.